tests/unit/customizations/s3/test_filters.py

```diff
@@ -202,6 +202,21 @@ def test_create_root_s3_no_dir_op(self):
         filtered = list(s3_filter.call(s3_files))
         self.assertEqual(len(filtered), 0)
 
+    def test_create_filter_s3_to_s3(self):
+        source = 'bucket/'
+        destination = 'bucket-2/'
+        pattern = '*'
+        parameters = {'filters': [['--exclude', pattern]],
+                      'dir_op': True,
+                      'src': 's3://' + source,
+                      'dest': 's3://' + destination}
+        s3_filter = self.create_filter(parameters=parameters)
+        source_pattern = s3_filter.patterns[0][1]
+        destination_pattern = s3_filter.dst_patterns[0][1]
+        self.assertEquals(source_pattern, source + pattern)
+        self.assertEquals(destination_pattern, destination + pattern)
+
```

Comment:
It would also be nice to test that the filter is applied correctly as well. So maybe pass some file stat objects that represent what the bug was and show that the issue is fixed.